src/engine/controls/bpmcontrol.cpp

```diff
@@ -470,6 +470,11 @@ double BpmControl::calcSyncAdjustment(bool userTweakingSync) {
     double syncTargetBeatDistance = m_dSyncTargetBeatDistance.getValue();
```

Comment:
This can make use of a comment.
// This is the distance from the last beat in forward direction.  

I am not sure if it is a good idea to use the forward direction in a "beat_distance" control. 
We need here the  distance from the last beat in play direction. I think it can be expected that the "beat_distance" control does this natively. Otherwise we need here the play direction of the sync target.

Reply:
No, I think it should always use the forward direction, like this:

    |---|  -> 0.0/1.0
    ^   
    |---|  -> 0.75
     ^   
    |---|  -> 0.5
      ^   
    |---|  -> 0.25
       ^   
    |---|  -> 0.0/1.0
        ^

Reply:
What else do we use "beat_distance" for? 
If we need to know in every single place the direction as well to make use of this value, it is a good indicator that the current forward only implementation is too unhandy and error prone.

Reply:
My mapping uses it for the glowing ring around the jog wheels: https://github.com/mixxxdj/mixxx/commit/876cd47dd239e099335610b131c4bc6be57e2180

Reply:
It looks like your code is also broken in reverse direction an will benefit from the change. 
When I got it right it re-implements the "beat_active" control which will be fixed with https://github.com/mixxxdj/mixxx/pull/3608.

Reply:
It doesn't work correctly when playing backwards, but that's unrelated to beat_distance. The issue is that I just count up beat indices to get the beat in the bar because we don't have that yet. If we had something like `bar_distance` or a `beat_number_in_bar` CO, it would work perfectly.

If we suddenly change the beat distance value when the playback direction changes, it wouldn't work and it would be impossible to fix it at all, even if we had bar detection COs.

Reply:
This should be fixable by changing the getBeatDistance code in syncable.cpp.  Much like the half/double calculation, if we do all the conversion in the syncable code, the rest of the program can pretend that all the decks are moving forwards and it should work fine